src/java_bytecode/library/src/java/lang/Thread.java

```diff
@@ -0,0 +1,44 @@
+package java.lang;
+
+import org.cprover.CProver;
+
+public class Thread implements Runnable
+{
+  private Runnable target;
+  private long tid;
+  private volatile static long threadSeqNumber;
+
+  public Thread()
+  {
+    target = this;
+    tid = ++threadSeqNumber;
```

Comment:
The `Thread` class has been heavily modifyed wrt the original:

http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/6-b14/java/lang/Thread.java

I propose that we use the same modus operandi as in the Modelling team and copy paste the original and insert calls to `CProver.notModelled()` right at the beginning of every method.